### ISSUE TITLE
oci_pull reproducible seems to not do anything

### ISSUE THREAD

**Issue by joca-bt** [bazel-contrib/rules_oci#347]
When pulling an image with `oci_pull`, `reproducible = False` seems to be ignored as I still get the warning.

```
load("@rules_oci//oci:pull.bzl", "oci_pull")
oci_pull(
    name = "base_image",
    image = "...",
    reproducible = False,
)
```

```
> bazel run //package:image
WARNING: fetching from https://... without an integrity hash. The result will not be cached.
WARNING: registry responded with a manifest that has schemaVersion=1. Usually happens when fetching from a registry that requires `Docker-Distribution-API-Version` header to be set.
Falling back to using `curl`. See https://github.com/bazelbuild/bazel/issues/17829 for the context.
WARNING: fetching from https://... without an integrity hash. The result will not be cached.
WARNING: for reproducible builds, a digest is recommended.
Either set 'reproducible = False' to silence this warning,
or run the following command to change oci_pull to use a digest:


buildozer 'set digest "sha256:2e45762ac9ba8ac55ed74d74153cc7663f5941b1fcdb35b6895738e9b061d5cd"' 'remove tag' 'remove platforms'  WORKSPACE:base_image

WARNING: fetching from https://... without an integrity hash. The result will not be cached.
WARNING: registry responded with a manifest that has schemaVersion=1. Usually happens when fetching from a registry that requires `Docker-Distribution-API-Version` header to be set.
Falling back to using `curl`. See https://github.com/bazelbuild/bazel/issues/17829 for the context.
WARNING: fetching from https://... without an integrity hash. The result will not be cached.
INFO: Analyzed target //package:image (12 packages loaded, 67 targets configured).
INFO: Found 1 target...
Target //package:image up-to-date:
  .bazel/bin/package/image/tarball.tar
INFO: Elapsed time: 21.672s, Critical Path: 5.86s
INFO: 24 processes: 17 internal, 4 linux-sandbox, 3 local.
INFO: Build completed successfully, 24 total actions
INFO: Running command line: .bazel/bin/package/image.sh
5d9749af06dc: Loading layer [==================================================>]    126MB/126MB
077f0bf4fc5e: Loading layer [==================================================>]  1.098kB/1.098kB
2700eebebe7e: Loading layer [==================================================>]     631B/631B
3cd7f5fab13a: Loading layer [==================================================>]  199.8MB/199.8MB
c3f3ffeae298: Loading layer [==================================================>]    136MB/136MB
6e6d4ee33b10: Loading layer [==================================================>]  2.255kB/2.255kB
Loaded image: bazel/package:image
```

**Comment by alexeagle** [comment]
Ha, buildifier was trying to tell us the same thing, the parameter appears unused:

<img width="1157" alt="Screenshot 2023-09-13 at 7 07 50 PM" src="https://github.com/bazel-contrib/rules_oci/assets/47395/b608f41f-61b0-4807-9bf3-c78adad641a2">
